m4/ac_path_kernel_source.m4

```diff
@@ -101,7 +101,9 @@ AC_DEFUN([AC_KERNEL_CHECKS],
                            -e s/sparc32.*/sparc/ \
                            -e s/sparc64.*/sparc/ \
                            -e s/s390x/s390/)
+  save_CFLAGS="$CFLAGS"
   save_CPPFLAGS="$CPPFLAGS"
+  CFLAGS="$CFLAGS -Wno-error"
```

Comment:
why needed?

Reply:
This is to override any passed `-Werror`. CFLAGS is meant for userspace, but it's also used here by gcc to check for kernel features. This feature check can fail as some warning can appear with -Wall (or even without).

Reply:
can we define -Werror only for userspace?
for example, move it to the relevant Makefile instead of adding it to CI yml files